src/nbl/video/COpenGLCommandPool.h

```diff
@@ -0,0 +1,118 @@
+#ifndef __NBL_C_OPENGL_COMMAND_POOL_H_INCLUDED__
+#define __NBL_C_OPENGL_COMMAND_POOL_H_INCLUDED__
+
+#include "nbl/video/IGPUCommandPool.h"
+
+namespace nbl {
+namespace video
+{
+
+class COpenGLCommandPool : public IGPUCommandPool
+{
+public:
+    using IGPUCommandPool::IGPUCommandPool;
+
+#define COMMAND_TYPES_LIST \
```

Comment:
turn this into an X-macro or something

Reply:
what is X-macro

Reply:
a way to use includes and macros together, such a list of types would be in a little include file and get included all over the place, google it, its an actual term

Reply:
im using NBL_FOREACH for this